lib/experimental/interactivity-api/directive-processing.php

```diff
@@ -56,6 +58,69 @@ function gutenberg_process_directives_in_root_blocks( $block_content, $block ) {
 }
 add_filter( 'render_block', 'gutenberg_process_directives_in_root_blocks', 10, 2 );
 
+/**
+ * Creates a stack of interactive block children.
+ *
+ * @param array    $parsed_block The parsed block.
+ * @param array    $source_block The source block.
+ * @param WP_Block $parent_block The parent block.
+ */
+function gutenberg_mark_interactive_block_children( $parsed_block, $source_block, $parent_block ) {
+	if (
+			isset( $parent_block ) &&
+			isset( $parent_block->block_type->supports['interactivity'] ) &&
+			$parent_block->block_type->supports['interactivity']
+		) {
+		WP_Directive_Processor::mark_children_of_interactive_block( $source_block );
+	}
+	return $parsed_block;
+}
+add_filter( 'render_block_data', 'gutenberg_mark_interactive_block_children', 100, 3 );
+
+/**
+ * Add a marker indicating if the block is interactive or not.
+ * core/interactivity-wrapper if it is interactive.
+ * core/non-interactivity-wrapper if it is not interactive.
+ *
+ * @param string   $block_content The block content.
+ * @param array    $block The full block, including name and attributes.
+ * @param WP_Block $block_instance The block instance.
+ */
+function gutenberg_mark_block_interactivity( $block_content, $block, $block_instance ) {
+	if (
+			isset( $block_instance->block_type->supports['interactivity'] ) &&
+			$block_instance->block_type->supports['interactivity']
+		) {
+		// Mark interactive blocks so we can process them later.
+		/**
+		 * Debugging purposes only. Nested comments are not allowed.
+		 * We wrap a hidden textarea to save the block content delimited
+		 * by comments so we can later process it.
+		 */
+		return sprintf(
+			'<div data-wp-delimiter="interactivity-wrapper-start" style="display:none"></div>%s<div data-wp-delimiter="interactivity-wrapper-end" style="display:none"></div>',
```

Comment:
_Just a note for anyone looking at this PR: this will be replaced with `get_comment_delimited_block_content`._

Reply:
Already done.